aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/util/hashing/CrcNative.kt

```diff
@@ -21,6 +24,10 @@ internal class Crc(val checksumFn: AwsChecksumsCrcFunction) : HashFunction {
     private var crc = 0U
 
     override fun update(input: ByteArray, offset: Int, length: Int) {
+        if (input.isEmpty() || length == 0) {
+            return
+        }
+
```

Comment:
**Question**: Why is this necessary? What breaks if we just pass an empty blob or a 0 length down to CRT?

Reply:
The request would never make it to CRT, the following code snippet fails with an `ArrayIndexOutOfBoundsException` exception:

```
        val offsetInput = input.usePinned {
            it.addressOf(offset)
        }
```